site/app/libraries/database/DatabaseQueries.php

```diff
@@ -107,8 +107,19 @@ public function insertSubmittyUser(User $user) {
         throw new NotImplementedException();
     }
 
-    public function loadThreads($announcements){
-        $this->course_db->query("SELECT * FROM threads WHERE deleted = false and pinned = ? ORDER BY id DESC", array($announcements));
+
+    public function loadThreads($announcements, $category_id){
+    	if($category_id === -1) {
+      		$this->course_db->query("SELECT t.*, e.category_id as category_id, w.category_desc FROM threads t, thread_categories e, categories_list w WHERE deleted = false and pinned = ? and t.id = e.thread_id and e.category_id = w.category_id ORDER BY t.id DESC", array($announcements));
+    	} else {
+
+    		$this->course_db->query("SELECT t.*, e.category_id as category_id, w.category_desc FROM threads t, thread_categories e, categories_list w WHERE deleted = false and pinned = ? and w.category_id = ? and t.id = e.thread_id and e.category_id = w.category_id ORDER BY t.id DESC", array($announcements, $category_id));
+    	}
+        return $this->course_db->rows();
+    }
```

Comment:
Why is this one function? Both queries are subtly different (with no obvious way to resolve them together at the moment) and don't both use all parameters. Just have a function `loadAnnouncements` and `loadThreads` or whatever.